schema/gen/go/genList.go

```diff
@@ -29,7 +29,74 @@ func (g listGenerator) EmitNativeType(w io.Writer) {
 }
 
 func (g listGenerator) EmitNativeAccessors(w io.Writer) {
-	// FUTURE: come back to this -- surely something nice can be done here.
+	// Generate a speciated Lookup as well as LookupMaybe method.
+	// The Lookup method returns nil in case of *either* an out-of-range/absent value or a null value,
+	//  and so should only be used if the list type doesn't allow nullable keys or if the caller doesn't care about the difference.
+	// The LookupMaybe method returns a MaybeT type for the list value,
+	//  and is needed if the list allows nullable values and the caller wishes to distinguish between null and out-of-range/absent.
+	// (The Lookup method should be preferred for lists that have non-nullable keys, because LookupMaybe may incur additional costs;
+	//   boxing something into a maybe when it wasn't already stored that way costs an alloc(!),
+	//    and may additionally incur a memcpy if the maybe for the value type doesn't use pointers internally).
+	doTemplate(`
+		func (n *_{{ .Type | TypeSymbol }}) Lookup(idx int) {{ .Type.ValueType | TypeSymbol }} {
```

Comment:
thought on naming: i might more naturally think of this at `Get` rather than `Lookup`

Reply:
[I thought about this for a surprisingly long time](https://github.com/ipld/go-ipld-prime/issues/22). At this point, I'd say either all the methods on the `Node` interface turn into `Get*` and this goes with them, or, none of them do.  Having the gen'd methods with concrete types have a totally different name prefix than the generic interface methods was the one thing that was definitely a mistake.